### PR TITLE
jira -> user workaround table

### PR DESCRIPTION
**Pull request by lwsanty**
related to [[DEV-1720]]

Signed-off-by: lwsanty <lwsanty@gmail.com>

[DEV-1720]: https://athenianco.atlassian.net/browse/DEV-1720

### CODE REVIEW

File: jira/database/database_user_pg.go [hunk 1]
```diff
@@ -63,6 +63,45 @@ func scanUser(sc dbs.Scanner) (User, error) {
 	}, err
 }
 
+func (db *pgDatabase) CreateUserWorkaround(ctx context.Context, uw *UserWorkaround) error {
+	_, err := db.ExecContext(ctx, `INSERT INTO
+		jira.user_workaround(acc_id, is_workaround)
+		VALUES($1, $2)
+		ON CONFLICT(acc_id) DO UPDATE SET is_workaround = $2`,
```
Comment (dennwc) [inline jira/database/database_user_pg.go:70]: This should actually be something like `OLD.is_workaround OR NEW.is_workaround` (pseudocode).

File: jira/database/database.go [hunk 2]
```diff
@@ -86,6 +87,8 @@ type Database interface {
 	CreateUser(ctx context.Context, u *User) error
 	GetUser(ctx context.Context, accID AccID, id UserID) (*User, error)
 	ListUsers(ctx context.Context, accID AccID) (UserIter, error)
+	CreateUserWorkaround(ctx context.Context, uw *UserWorkaround) error
```
Comment (dennwc) [inline jira/database/database.go:90]:
```suggestion
	CreateUserWorkaround(ctx context.Context, accID AccID, workaround bool) error
```
Not worth a separate struct.

File: sql/migrations/115_jira_user_workaround.up.sql [hunk 3]
```diff
@@ -0,0 +1,8 @@
+-- DEV-1720
+BEGIN;
+CREATE TABLE jira.user_workaround (
+    acc_id bigint NOT NULL,
+    is_workaround bool,
```
Comment (dennwc) [inline sql/migrations/115_jira_user_workaround.up.sql:5]: Should either be `NOT NULL` or have a default.

File: jira/database/database.go (hunk 2 above)
Comment (lwsanty) [reply jira/database/database.go]: Done.

File: sql/migrations/115_jira_user_workaround.up.sql (hunk 3 above)
Comment (lwsanty) [reply sql/migrations/115_jira_user_workaround.up.sql]: Done.

File: jira/database/database_user_pg.go (hunk 1 above)
Comment (lwsanty) [reply jira/database/database_user_pg.go:70]: agreed to keep this one since we run it only in handle users and only for the fist page.

**Review by dennwc — APPROVED** [review]

**Pull request merged**